### PR TITLE
fix: add id and name fields in data sources should be optional and co…

### PR DESCRIPTION
**Pull request by cristiGuranIonos**
…mputed

## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: ionoscloud/data_source_dbaas_inmemorydb_replicaset.go [hunk 1]
```diff
@@ -30,6 +31,7 @@ func dataSourceDBaaSInMemoryDBReplicaSet() *schema.Resource {
 				Type:        schema.TypeString,
 				Description: "The display name of the InMemoryDB Replicaset.",
 				Optional:    true,
+				Computed:    true,
```
Comment (adeatcu-ionos) [inline ionoscloud/data_source_dbaas_inmemorydb_replicaset.go:34]: @cristiGuranIonos why is `location` a `Computed` field? From what I know, the `location` is never returned in the response of the API.

File: ionoscloud/data_source_template_test.go [hunk 2]
```diff
@@ -22,37 +22,31 @@ func TestAccDataSourceTemplate(t *testing.T) {
...
+					resource.TestCheckResourceAttr(templateName, "cores", "2"),
+					resource.TestCheckResourceAttr(templateName, "ram", "4096"),
+					resource.TestCheckResourceAttr(templateName, "storage_size", "120"),
 				),
 			},
 			{
-				Config: testAccDataSourceTemplateCores,
+				Config: testAccDataSourceTemplateCoresRam,
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(templateName, "name", "CUBES XL"),
```
Comment (adeatcu-ionos) [inline ionoscloud/data_source_template_test.go:34]: @cristiGuranIonos why are we removing these checks?

File: ionoscloud/resource_server_test.go [hunk 3]
```diff
@@ -1088,7 +1083,6 @@ resource ` + constant.ServerResource + ` ` + constant.ServerTestResource + ` {
   cores = 1
   ram = 1024
   availability_zone = "ZONE_1"
-  cpu_family = "INTEL_XEON"
```
Comment (adeatcu-ionos) [inline ionoscloud/resource_server_test.go:1091]: @cristiGuranIonos Is this `cpu_family` not available anymore? Why are we removing it?

File: ionoscloud/data_source_template_test.go (hunk 2 above)
Comment (cristiGuranIonos) [reply ionoscloud/data_source_template_test.go:34]: easier to keep the test up-to-date when they change. they're tested in the first step, so they definitely work.

File: ionoscloud/resource_server_test.go (hunk 3 above)
Comment (cristiGuranIonos) [reply ionoscloud/resource_server_test.go:1091]: keeps the tests easier to maintain. Either a cpu family is removed, it's not available or changes. This allows the server to choose it's own cpu family.

File: ionoscloud/data_source_dbaas_inmemorydb_replicaset.go (hunk 1 above)
Comment (cristiGuranIonos) [reply ionoscloud/data_source_dbaas_inmemorydb_replicaset.go:34]: I think this is dispay_name, not location. Location should be optional.

Comment (adeatcu-ionos) [reply ionoscloud/data_source_dbaas_inmemorydb_replicaset.go:34]: @cristiGuranIonos sorry, git diff tricked me, it wasn't an expanded tab and I thought you marked `location` as `Computed`.

**Review by adeatcu-ionos — APPROVED** [review]

**Pull request merged**